encoding/codecv2/codecv2.go

```diff
@@ -0,0 +1,618 @@
+package codecv2
+
+/*
+#cgo LDFLAGS: -L../../rs -lm -ldl -lscroll_zstd
+#include <stdlib.h>
+#include "../../rs/rs_zstd.h"
```

Comment:
it'll be better that generate the `rs_zstd.h` and `libzktrie.so` to `encoding` folder.